modules/cli/src/main/scala/scala/cli/commands/migratedirectives/MigrateDirectives.scala

```diff
@@ -0,0 +1,104 @@
+
+object MigrateDirectives extends ScalaCommand[MigrateDirectivesOptions] {
+  override def group                   = "Main"
+  override def scalaSpecificationLevel = SpecificationLevel.RESTRICTED
+  override def sharedOptions(options: MigrateDirectivesOptions): Option[SharedOptions] =
+    Some(options.shared)
+
+  override def runCommand(
+    options: MigrateDirectivesOptions,
+    args: RemainingArgs,
+    logger: Logger
+  ): Unit = {
+    val buildOptions = buildOptionsOrExit(options)
+    val inputs       = options.shared.inputs(args.remaining, () => Inputs.default()).orExit(logger)
+    val preprocessors = Sources.defaultPreprocessors(
+      buildOptions.scriptOptions.codeWrapper.getOrElse(CustomCodeWrapper),
+      buildOptions.archiveCache,
+      buildOptions.internal.javaClassNameVersionOpt,
+      () => buildOptions.javaHome().value.javaCommand
+    )
+    val (_, preprocessedSources) = CrossSources.allInputsAndPreprocessedSources(
+      inputs,
+      preprocessors,
+      options.shared.logger
+    ).orExit(logger)
+
+    val projectFilePath = inputs.elements.projectSettingsFiles.headOption match
+      case Some(s) => s.path
+      case _       => inputs.workspace / Constants.projectFileName
+    val preprocessedWithUsingDirs = preprocessedSources.filter(_.directivesPositions.isDefined)
+
+    def splitCodeAndDirectives(
+      contentLines: Seq[String],
+      positions: DirectivesPositions
+    ): (String, String) = {
+      val endLine   = positions.scope.endPos._1
+      val endColumn = positions.scope.endPos._2
+      val (indexedDirectives, indexedCode) =
+        contentLines.zipWithIndex.partition((_, idx) =>
+          idx < endLine || (idx == endLine && endColumn > 0)
+        )
+      val code       = indexedCode.map(_._1)
+      val directives = indexedDirectives.map(_._1)
+      (code.mkString("\n"), directives.mkString("\n"))
+    }
+
+    preprocessedWithUsingDirs.length match
+      case 0 => logger.message("No using directives have been found")
+      case 1 => logger.message("All using directives are already in one file")
+      case _ =>
+        val directivesSeq = preprocessedWithUsingDirs
+          .filter(_.scopePath != ScopePath.fromPath(projectFilePath))
+          .map { source =>
+            source match
+              case file: OnDisk =>
+                val contentLines = os.read.lines(file.path)
+                val result = file.directivesPositions match
+                  case Some(positions) =>
+                    val (code, directives) = splitCodeAndDirectives(contentLines, positions)
+                    logger.message(s"Deleting directives from ${file.path}...")
+                    os.write.over(file.path, code.getBytes(StandardCharsets.UTF_8))
+                    Seq(directives)
+                  case _ => Nil
+                result
+
+              case file: InMemory =>
+                val contentLines = file.code.split("\n").toSeq
```

Comment:
seems like the only difference between how `InMemory` and `OnDisk` elements are handled is how the `contentLines` are extracted... the rest seems like duplicated code, should be abstracted